super-ddf.c

```diff
@@ -877,30 +881,178 @@ static void *load_section(int fd, struct ddf_super *super, void *buf,
 	return buf;
 }
 
-static int load_ddf_headers(int fd, struct ddf_super *super, char *devname)
+
+/*
+ * Search for DDF_HEADER_MAGIC in the last 32MB of the device
+ *
+ * According to the specification, the Anchor Header should be placed at
+ * the end of the disk. However,some widely used RAID hardware, such as
+ * LSI and PERC, do not position it within the last 512 bytes of the disk.
+ *
+ */
+static int search_for_ddf_headers(int fd, char *devname,
+				  unsigned long long *out)
 {
 	unsigned long long dsize;
+	unsigned long long search_start;
```

Comment:
variables should be declared in reverse Christmas tree convention:
```c
	unsigned long long search_start;
	unsigned long long search_end;
	size_t bytes_block_to_read;
	unsigned long long dsize;
	unsigned long long pos;
	int bytes_current_read;
	be32 *magic_ptr = NULL;
	void *buffer = NULL;

	int result = 0;
```
This applied to the all change. It is not critical but I have to point you that.